javaslang-benchmark/src/test/java/javaslang/MemoryUsage.java

```diff
@@ -1,37 +1,62 @@
 package javaslang;
 
-import javaslang.collection.CharSeq;
-import javaslang.collection.LinkedHashMultimap;
-import javaslang.collection.List;
-import javaslang.collection.Multimap;
+import javaslang.collection.*;
+
+import java.text.DecimalFormat;
+import java.util.regex.Pattern;
 
 import static com.carrotsearch.sizeof.RamUsageEstimator.humanReadableUnits;
 import static com.carrotsearch.sizeof.RamUsageEstimator.sizeOf;
-import static javaslang.BenchmarkPerformanceReporter.DECIMAL_FORMAT;
+import static java.lang.Math.max;
 
 public class MemoryUsage {
-    private static Multimap<Integer, String> memoryUsages = LinkedHashMultimap.withSeq().empty(); // if forked, this will be reset every time
+    private static final DecimalFormat FORMAT = new DecimalFormat("#,##0");
+    private static Map<Integer, LinkedHashSet<Tuple3<CharSeq, CharSeq, CharSeq>>> memoryUsages = TreeMap.empty(); // if forked, this will be reset every time
 
     /** Calculate the occupied memory of different internals */
     static void printAndReset() {
-        for (int size : memoryUsages.keySet()) {
-            System.out.println(String.format("\nfor `%d` elements", size));
-            for (String usages : memoryUsages.get(size).get()) {
-                System.out.println(usages);
+        for (Tuple2<Integer, LinkedHashSet<Tuple3<CharSeq, CharSeq, CharSeq>>> entry : memoryUsages) {
+            final Tuple3<Integer, Integer, Integer> columnSizes = columnSized(entry._1);
+            System.out.println(String.format("\nfor `%d` elements", entry._1));
+            for (Tuple3<CharSeq, CharSeq, CharSeq> stats : entry._2) {
+                final String format = String.format("  %s → %s bytes - %s",
+                        stats._1.padTo(columnSizes._1, ' '),
+                        stats._2.leftPadTo(columnSizes._2, ' '),
+                        stats._3.leftPadTo(columnSizes._3, ' ')
+                );
+                System.out.println(format);
             }
         }
 
         memoryUsages = memoryUsages.take(0); // reset
     }
+    private static Tuple3<Integer, Integer, Integer> columnSized(int size) {
+        return memoryUsages.get(size)
```

Comment:
it would make sense to make tuples support better for these situations ... or should I use a `Seq` here instead?